docs/upgrade.html

```diff
@@ -67,6 +67,7 @@ <h4><a id="upgrade_1_2_0" href="#upgrade_1_2_0">Upgrading from 0.8.x, 0.9.x, 0.1
 <h5><a id="upgrade_120_notable" href="#upgrade_120_notable">Notable changes in 1.2.0</a></h5>
 <ul>
     <li><a href="https://cwiki.apache.org/confluence/x/oYtjB">KIP-186</a> increases the default offset retention time from 1 day to 7 days. This makes it less likely to "lose" offsets in an application that commits infrequently. It also increases the active set of offsets and therefore can increase memory usage on the broker. Note that the console consumer currently enables offset commit by default and can be the source of a large number of offsets which this change will now preserve for 7 days instead of 1. You can preserve the existing behavior by setting the broker config <code>offsets.retention.minutes</code> to 1440.</li>
+    <li><a href="https://cwiki.apache.org/confluence/x/AZQ7B">KIP-174</a> deprecates the <code>internal.key.converter</code> and <code>internal.value.converter</code> configurations for Kafka Connect, and establishes <code>org.apache.kafka.connect.json.JsonConverter</code> with the <code>schemas.enable</code> property set to <code>false</code> as the default value for both.</li>
```

Comment:
Might be good to explain whether people need to worry about this. And I wonder if it'd be worthwhile to include the configuration properties that are now used by default so it's easier for people to see whether something has changed.

Reply:
> Might be good to explain whether people need to worry about this.

Agreed, although I'm actually uncertain on the steps they should take if it turns out they do need to worry. Is it even possible to transition from one internal converter to another without wiping out the various topics (offests, config, storage) used by the Connect framework? What do we recommend if it turns out that's not actually possible?

> And I wonder if it'd be worthwhile to include the configuration properties that are now used by default so it's easier for people to see whether something has changed.

Isn't that already done with the `establishes <code>org.apache.kafka.connect.json.JsonConverter</code> with the <code>schemas.enable</code> property set to <code>false</code> as the default value for both.` part?

Reply:
Think of it this way: you're a user, you see this warning, and you want to know if you need to do something? You look in the documentation, and the documentation says that if your worker config contains these lines you can simply remove them now or leave them alone (and maybe get warnings in the future).

I don't think properties embedded in a paragraph are good enough. However, if you have a code block with the four lines that are equivalent to what Connect now does by default, then people will quickly recognize that's what they have and, per the documentation, conclude they can remove these lines from their worker configs without a change in behavior.

Reply:
I'm concerned about what we tell anyone who's using a different internal converter and/or one with `schemas.enable=true` (although I suspect that at least in the latter case just setting the value to `false` might work). If we're planning on completely removing these configurations in a future release, we should let people know now how they can deal with these changes and upgrade when the time comes.